docs/internal/inheritance.md

```diff
@@ -0,0 +1,45 @@
+# Inheritance
+
+In Verona, class inheritance is not the same as sub-typing.
+A type is a sub-type of another if explicitly declared by the programmer.
+
+Example:
+```
+  interface Foo {
+    f(...);
+  };
+  class Bar : Foo {
+    f(...) { ... }
+  };
+```
+
+Here, `Bar` is a sub-type of `foo`, explicit in the declaration `Bar : Foo`.
+However, `Bar` also has implementation inheritance, because it implements `Foo`'s `f` function.
+
+## Capabilities
+
+Like classes, capabilities can be sub-types of each other.
+This is used to allow objects with different capabilities to reference each other.
+
+Example:
+```
+  let x : Foo & mut = ...;
+
+  // This is fine, as `readonly` is a sub-type of `mut`
```

Comment:
You got subtyping the wrong way round, `mut` would be subtype of `readonly`.

Reply:
Ah, yes! I always mix that up. I instinctively see `readonly` as more restrictive, and therefore "less". 

But here, `mut` can be used as a `readonly` (by not changing it) but not the other way round, so `mut` is a subtype of `readonly`.